### DISCUSSION TITLE
[Question][Hw2] Hint on where to start looking for failing tests

### DISCUSSION THREAD

**Discussion by m-spitfire** [prosyslab-classroom/cs524-program-analysis#293] [Q&A] [answered]
Name: Murad Bashirov

I've been working on the assignment, and I got 96/98 test cases passing. However, I can't seem to find the bug that makes thlss two test cases fail. I've tried everything, I don't think my implementation is wrong from what's been described in the assignment. My first suspect was something related to i32 type, but I keep everything strictly in the bounds of 32-bit integer and raise exception whenever overflow happens, so I don't think that's the issue.

What could be the issue? How can I know what's wrong with my code? Because C doesn't have any formal specification, and we're only working on subset of C, it's not easy to find buggy part that makes tests fail.

**Comment by KihongHeo** [comment, marked as answer]
Hi, I appreciate your hard working.
First of all, this language is not C; we do not expect you to implement C semantics. Please focus on the provided document.

Also, please note that the leaderboard is provided for your convenience, not for debugging. TAs can evaluate all your homework after the deadline. But they set up the board for immediate feedback, avoiding mis-submission, etc. So, unless there is an error in the board, everything should be done by yourself.

**Comment by m-spitfire** [comment]
I see. Thanks for the response.
